src/main.cpp

```diff
@@ -40,270 +26,264 @@ void setup(){
 
   LCD.begin(16,2) ;
 
-  Serial.begin(9600) ;
+  Serial.begin(115200) ;
 
   RCServo0.attach(SERVO_0());
   RCServo1.attach(SERVO_1());
   RCServo2.attach(SERVO_2());
   RCServo3.attach(SERVO_3());
-
-  left_surface = false;  // TODO read the switch
+  
+  left_surface = true;  // TODO read the switch
   // left_surface = digitalRead(LEFT_RIGHT_SWITCH());
+
+  hardware::Qrd qrd;
+  hardware::Beacon beacon;
+  hardware::Encoder l_encoder(0);
+  hardware::Encoder r_encoder(1, true);
+
+  hardware::EncoderMotor l_motor(L_MOTOR(), l_encoder, true);
+  hardware::EncoderMotor r_motor(R_MOTOR(), r_encoder);
+
+  sequences::Claw claw;
+  sequences::Drivetrain driver(l_motor, r_motor, qrd);
+  sequences::Platform platform(driver);
+
+  stages::Gate gate(driver, beacon, r_encoder); // TODO change encoder dynamically
+  stages::Pickup pickup(qrd, r_encoder, claw, driver);
+  stages::Zipline zipline(driver, platform, beacon, r_encoder, qrd);
+
   claw.fold(left_surface);
   delay(1000);
 
-  // claw.raise(left_surface ? sequences::RIGHT_CLAW : sequences::LEFT_CLAW);
-  // while(!claw.loop()) delay(100);
-  // claw.raise(left_surface ? sequences::LEFT_CLAW : sequences::RIGHT_CLAW);
-  // while(!claw.loop()) delay(100);
+  unsigned long waitUntil = millis() + loop_delay;
+  
+  enableEncoderInterrupts();
```

Comment:
move line 58 to 60 to after `Insert extra variable here`. This way delays in the intialization won't affect things